File: articles/heuristics-and-cognitive-biases/python/run_all_bias_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Heuristics and Cognitive Biases."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "anchoring_bias_model.py",
    "availability_bias_model.py",
    "representativeness_bias_model.py",
    "confirmation_bias_diagnostics.py",
    "confidence_distortion_analysis.py",
    "calibration_error_scoring.py",
    "debiasing_review_queue.py",
    "decision_record_exporter.py",
    "heuristics_cognitive_biases_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python bias workflows completed.")


if __name__ == "__main__":
    main()

File: articles/heuristics-and-cognitive-biases/r/run_all_bias_workflows.R
# run_all_bias_workflows.R
# Run all R workflows for Heuristics and Cognitive Biases.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "heuristics_cognitive_biases_workflow.R",
  "bias_profile_summary.R",
  "calibration_error_tables.R",
  "confidence_gap_reports.R",
  "domain_bias_diagnostics.R",
  "debiasing_review_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R bias workflows completed.")
